server/src/test/java/io/crate/execution/engine/aggregation/impl/CountAggregationTest.java

```diff
@@ -75,6 +121,215 @@ public void test_function_implements_doc_values_aggregator_for_string_based_type
         }
     }
 
+    private void test_count_on_object_with_not_null_immediate_child(DataType<?> childType, Class<?> expectedAggregatorClass) {
+        Reference notNullImmediateChild = new Reference(
+            new ReferenceIdent(null, new ColumnIdent("top_level_object", "not_null_subcol")),
+            RowGranularity.DOC,
+            childType,
+            0,
+            null);
+        Reference countedObject = new Reference(
+            new ReferenceIdent(null, new ColumnIdent("top_level_object")),
+            RowGranularity.DOC,
+            ObjectType.builder().setInnerType(notNullImmediateChild.column().leafName(), notNullImmediateChild.valueType()).build(),
+            0,
+            null
+        );
+        DocTableInfo sourceTable = mock(DocTableInfo.class);
+        when(sourceTable.notNullColumns()).thenReturn(List.of(notNullImmediateChild.column()));
+        when(sourceTable.getReference(eq(notNullImmediateChild.column()))).thenReturn(notNullImmediateChild);
+
+        assertHasDocValueAggregator(List.of(countedObject), sourceTable, expectedAggregatorClass);
+        verify(sourceTable, times(1)).notNullColumns();
+        verify(sourceTable, times(1)).getReference(eq(notNullImmediateChild.column()));
+    }
+
+    @Test
+    public void test_count_on_object_with_not_null_immediate_child() {
+        var countedObjectToExpectedAggregatorMap = Map.ofEntries(
+            Map.entry(DataTypes.BYTE, SortedNumericDocValueAggregator.class),
+            Map.entry(DataTypes.TIMESTAMPZ, SortedNumericDocValueAggregator.class),
+            Map.entry(DataTypes.GEO_POINT, SortedNumericDocValueAggregator.class),
+            Map.entry(DataTypes.IP, BinaryDocValueAggregator.class),
+            Map.entry(DataTypes.STRING, BinaryDocValueAggregator.class)
+        );
+        for (var e : countedObjectToExpectedAggregatorMap.entrySet()) {
+            test_count_on_object_with_not_null_immediate_child(e.getKey(), e.getValue());
+        }
+    }
+
+    @Test
+    public void test_count_on_object_with_not_null_grand_child() {
```

Comment:
Would be nice to rename the test methods to clarify the test scenario a bit more.

Reply:
renamed, `test_count_on_object_with_deeper_not_null_subcolumn_uses_DocValueAggregator`